app/controllers/AddressWithoutIdController.scala

```diff
@@ -72,36 +67,34 @@ class AddressWithoutIdController @Inject() (
 
   def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData.apply andThen requireData).async {
     implicit request =>
-      val registeringAsBusiness: Boolean =
-        request.userAnswers.get(WhatAreYouRegisteringAsPage) match { //ToDo defaulting to registering for business change when paths created if necessary
-          case Some(RegistrationTypeBusiness) => true
-          case _                              => false
-        }
-      val countries = getFormWithCountries(registeringAsBusiness)
-      val form      = formProvider(countries)
+      implicit val registeringAsBusiness  = getRegisteringAsBusiness()
+      implicit val countryListFactoryImpl = countryListFactory
 
-      render(mode, request.userAnswers.get(AddressWithoutIdPage).fold(form)(form.fill), registeringAsBusiness, countries).map(Ok(_))
+      SomeInformationIsMissing.isMissingCountryList {
+        countries =>
+          val form = formProvider(countries)
+          render(mode, request.userAnswers.get(AddressWithoutIdPage).fold(form)(form.fill), registeringAsBusiness, countries).map(Ok(_))
+      }
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData.apply andThen requireData).async {
     implicit request =>
-      val registeringAsBusiness: Boolean =
-        request.userAnswers.get(WhatAreYouRegisteringAsPage) match { //ToDo defaulting to registering for business change when paths created if necessary
-          case Some(RegistrationTypeBusiness) => true
-          case _                              => false
-        }
-      val countries = getFormWithCountries(registeringAsBusiness)
+      implicit val registeringAsBusiness  = getRegisteringAsBusiness()
+      implicit val countryListFactoryImpl = countryListFactory
 
-      formProvider(countries)
-        .bindFromRequest()
-        .fold(
-          formWithErrors => render(mode, formWithErrors, registeringAsBusiness, countries).map(BadRequest(_)),
-          value =>
-            for {
-              updatedAnswers <- Future.fromTry(request.userAnswers.set(AddressWithoutIdPage, value))
-              _              <- sessionRepository.set(updatedAnswers)
-            } yield Redirect(navigator.nextPage(AddressWithoutIdPage, mode, updatedAnswers))
-        )
+      SomeInformationIsMissing.isMissingCountryList {
```

Comment:
@mbiczak This cannot be some information missing it is a blocker, if a country list is not found, the user cannot proceed. It is Technical difficulties case